File: client/src/main/java/su/plo/voice/client/config/overlay/OverlayStyle.java
package su.plo.voice.client.config.overlay;

import lombok.Getter;
import su.plo.lib.api.chat.MinecraftTextComponent;

public enum OverlayStyle {

    NAME_SKIN("gui.plasmovoice.overlay.style.name_skin"),
    SKIN("gui.plasmovoice.overlay.style.skin"),
    NAME("gui.plasmovoice.overlay.style.name");

    @Getter
    private final MinecraftTextComponent translatable;

    OverlayStyle(String key) {
        this.translatable = MinecraftTextComponent.translatable(key);
    }

    public boolean hasName() {
        return this == NAME_SKIN || this == NAME;
    }

    public boolean hasSkin() {
        return this == NAME_SKIN || this == SKIN;
    }

    public static OverlayStyle fromOrdinal(int ordinal) {
        switch (ordinal) {
            case 1:
                return OverlayStyle.SKIN;
            case 2:
                return OverlayStyle.NAME;
            default:
                return OverlayStyle.NAME_SKIN;
        }
    }
}

File: api/client/src/main/kotlin/su/plo/voice/api/client/config/overlay/OverlayStyle.kt
package su.plo.voice.api.client.config.overlay

import su.plo.slib.api.chat.component.McTextComponent

/**
 * Overlay style.
 */
enum class OverlayStyle(key: String) {

    NAME_SKIN("gui.plasmovoice.overlay.style.name_skin"),
    SKIN("gui.plasmovoice.overlay.style.skin"),
    NAME("gui.plasmovoice.overlay.style.name");

    val translatable = McTextComponent.translatable(key)

    val hasName: Boolean
        get() = this == NAME_SKIN || this == NAME

    val hasSkin: Boolean
        get() = this == NAME_SKIN || this == SKIN

    companion object {

        fun fromOrdinal(ordinal: Int): OverlayStyle {
            return when (ordinal) {
                1 -> SKIN
                2 -> NAME
                else -> NAME_SKIN
            }
        }
    }
}
